circle.yml

```diff
@@ -44,11 +46,17 @@ deployment:
   docker:
     branch: master
     commands:
-      # Build small deploy container for master branch.
       - sed "s/<EMAIL>/$DOCKER_EMAIL/;s/<AUTH>/$DOCKER_AUTH/" < "resources/deploy_templates/.dockercfg.template" > ~/.dockercfg
       - |
+          VERSION=$(git describe 2>/dev/null || git rev-parse --short HEAD 2>/dev/null)
```

Comment:
Yes, thought the same thing. Could just (morally) `//go:generate s/COCKROACH_VERSION/$(git describe)/ > version.go < version.go.template` or something.

Were you referring to this here?
http://stackoverflow.com/questions/11354518/golang-application-auto-build-versioning

The answer above likely precedes `go generate`, which looks like it was made for this sort of thing.